File: CS12b/PA3/DictionaryTest.java
//-----------------------------------------------------------------------------
// DictionaryTest.java
// Antonio Valenzuela
// anlvalen@ucsc.edu
// CS12B
// July 22, 2017
// PA3
// isolation tests for Dictionary.java 
//-----------------------------------------------------------------------------

public class DictionaryTest{
    public static void main(String[] args){
    Dictionary test = new Dictionary();
    boolean a; 
    String b; 
    int c;
    
    a = test.isEmpty();
    System.out.println("Correct output of isEmpty() is true. Actual output is: "+a);
    
    test.insert("z","2");
    test.insert("y","5");
    
    c = test.size();
    System.out.println("Correct output size is 2. Actual output is: "+c);
    
    b = test.lookup("y");
    System.out.println("Correct value of y is 5. Actual output is: "+b);
    
    //test.insert("z","1"); //trows DuplicateKeyException
    
    //test.delete("w"); //throws KeyNotFoundException
    
    a = test.isEmpty();
    System.out.println("Correct output of isEmpty is false. Actual output is: "+a);
    
    System.out.println(test);
    // expected output:
    // Correct output of isEmpty() is true. Actual output is: true
    // Correct output size is 2. Actual output is: 2
    // Correct value of y is 5. Actual output is: 5
    // Correct output of isEmpty is false. Actual output is: false
    // z 2
    // y 5
    }
}

File: CS12M/Lab 5/DictionaryTest.c
//-----------------------------------------------------------------------------
// DictionaryTest.c
// Antonio Valenzuela
// anlvalen@ucsc.edu
// CS12B
// Aug 6th, 2017
// Lab5
// Isolation testing for Dictionary.c (reference: DictionaryClient.c and
// DictionaryTest.java from PA3)
//-----------------------------------------------------------------------------
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

int main(int argc, char* argv[]){
    Dictionary A = newDictionary();
    char* k;
    char* v;
    char* word1[] = {"a","b","c","d","e"};
    char* word2[] = {"lets","get","ready","to","rumble"};
    int i;

    for(i=0; i<5; i++){
        insert(A, word1[i], word2[i]);
    }
    printf("%s\n", (isEmpty(A)?"true":"false"));
    printf("%d\n", size(A));
    //insert(A, "d", "fight");
    
    printDictionary(stdout, A);
    
    for(i=0; i<5; i++){
    k = word1[i];
    v = lookup(A, k);
    //v = lookup(B, k);
    printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
    }
    
    //delete(A, "one");
    //delete(B, "three");
    delete(A, "d");
    delete(A, "b");
    printf("%s\n", (isEmpty(A)?"true":"false"));
    printf("%d\n", size(A));
    printDictionary(stdout, A);

    for(i=0; i<5; i++){
        k = word1[i];
        v = lookup(A, k);
        printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
    }
    printf("%s\n", (isEmpty(A)?"true":"false"));
    printf("%d\n", size(A));
    makeEmpty(A);
    printf("%s\n", (isEmpty(A)?"true":"false"));
    printf("%d\n", size(A));
    freeDictionary(&A);
    
    return(EXIT_SUCCESS);
    }